src/core/net/ip6.cpp

```diff
@@ -504,6 +548,8 @@ ThreadError Ip6::HandleExtensionHeaders(Message &message, Header &header, uint8_
             break;
 
         case kProtoIp6:
+            ExitNow();
```

Comment:
Ok great, i was trying to reply to your below comment that disappear now :) Anyhow, i will make a recap here even if its not related now..

I process inner IPv6 header as you see only when we are subscribed to the destination address from outer IPv6 header, i think its intuitive. Do you think this behaviour is correct or we should process inner header always?